src/chocolatey.resources/helpers/functions/Get-Installed.ps1

```diff
@@ -0,0 +1,48 @@
+﻿## Get-Installed
+##############################################################################################################
+## Returns a list of installed applications.
+## Registry based replacement for "Get-WmiObject -Class Win32_Product".
+##############################################################################################################
+
+function Get-Installed {
+param(
+  [switch] $GWMICompatible = $false # You may use '-GWMICompatible' switch
+)
+
+  $items = @()
+
+  if(Test-Path "HKLM:SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall") {
+    $items += Get-ChildItem "HKLM:SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall"
+  }
+  if(Test-Path "HKLM:SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall") {
+    $items += Get-ChildItem "HKLM:SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall"
+  }
+
+  $productList = @()
+
+  foreach ($app in $items)
+  {
+    $propitems = $app | Foreach-Object {Get-ItemProperty $_.PsPath}
+    $w32item = New-Object System.Object
+    $w32item | Add-Member -type NoteProperty -Name "IdentifyingNumber" -Value $app.PsChildname
+    $w32item | Add-Member -type NoteProperty -Name "Name"    -Value $propitems.DisplayName
+    $w32item | Add-Member -type NoteProperty -Name "Vendor"  -Value $propitems.Publisher
+    $w32item | Add-Member -type NoteProperty -Name "Version" -Value $propitems.DisplayVersion

```

Comment:
This fix is mainly intended to be a replacement for Win32_Product which doesn't know about innosetup at all. Win32_Product works with Windows Installer (MSI).
Hooked Get-WmiObject function returns a list fully compatible with original function's output except an items order. Fields returned by Win32_Product:

- IdentifyingNumber
- Name
- Vendor
- **Version**
- Caption (allways the same as Name)

Get-Installed (called from Get_WmiObject) is able to return non-MSI applications and it does so when -GWMICompatible switch is **not** used. It had a switch with the opposite meaning in previous commits but this has been changed during a conversation (see outdated diffs).
Also InnoSetup MAY use DisplayVersion. Here is a part of output:

    IdentifyingNumber : Git_is1
    Name              : Git version 2.5.0
    Vendor            : The Git Development Community
    Version           : 2.5.0
    Caption           : Git version 2.5.0
    WindowsInstaller  : 
    UninstallString   : "C:\Program Files\Git\unins000.exe"

Why two functions?

1. Get-Installed can return MSI and non-MSI records. It may be used in packages installation/uninstallation scripts.
2. Get_WmiObject can return only MSI records. It transparently replaces Win32_Product calls with Get-Installed so there is no need to fix more than 60 existing (and may be some future) packages which use Win32_Product.